src/pages/Controllers/index.js

```diff
@@ -139,9 +140,23 @@ class Controllers extends Component {
     });
   };
 
+  removeControllerFromFavorites = (event, value, controller) => {
+    // Stop propagation from going to the next page
+    event.stopPropagation();
+    const { dispatch } = this.props;
+    // dispatch an action to favorite controller
+    dispatch({
+      type: 'user/removeControllerFromFavorites',
+      payload: controller,
+    });
+  };
+
   render() {
     const { controllers } = this.state;
     const { loadingControllers, favoriteControllers } = this.props;
+    const {
+      auth: { username },
```

Comment:
So we are using the presence of a "username" field in the auth part of the Redux store?  Should this be a method so that we don't to change so many other modules the implementation changes over time?

Reply:
Are you suggesting this to be a method like `getUsername()` which returns the username once it exists, null otherwise? Either way, "auth" is returned to this component as a prop, so I believe there should not be any harm in directly accessing it.

Ideally, I would like the "user" to be in the global state, passed down to lower components in the hierarchy as a context object, this reduces the redundancy of accessing "username" in every page component. We would just get the user field directly after that.

Reply:
> Are you suggesting this to be a method like `getUsername()` which returns the username once it exists, null otherwise? Either way, "auth" is returned to this component as a prop, so I believe there should not be any harm in directly accessing it.

True, "no harm", but in the abstract the code is really trying to say: "A user is requesting that the Dashboard marks a controller as favorite.  Is this user logged in or not?"

Instead of depending on the property existing to infer the user is logged in or not, why not make a method that returns the answer to the question explicitly and directly?  Perhaps `isLoggedInUser()`?

Then we can define that method to always answer that question `true`/`false`, and over time if we have to change the implementation details, the method does not have to change.

> Ideally, I would like the "user" to be in the global state, passed down to lower components in the hierarchy as a context object, this reduces the redundancy of accessing "username" in every page component. We would just get the user field directly after that.

Sure, that is an implementation detail, but the method should hide the knowledge of how to use the global state to answer a question.  We provide a method to answer the question directly.

Reply:
Right! Got the point!